chapter06/ard/ex1.pl

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use autodie;
+
+use 5.010;
+
+use Storable;
+
+my %data = ();
+my $datafile = "ex1.data";
+
+if ( -e -f $datafile ) {
```

Comment:
Ah, my bad (and good catch) - I added the `-f` as an afterthought when I decided that trying to read a directory was a bad idea!